LogfileIterator/LogfileIteratorExample.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2016 HARBRICK TECHNOLOGIES, INC
+ *
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/**
+ * \example LogfileIteratorExample.cpp
+ *
+ * PolySync Logfile CPP API Iterator Example.
+ *
+ * Shows how to use the Logfile API to iterate over records in a Logfile
+ * outside of the normal PolySync Replay time domain. This means that the
+ * Logfile iterator runs as fast as the CPU will allow.
+ *
+ * @file LogfileIteratorExample.cpp
+ * @brief LogfileIterator Source
+ *
+ */
+
+
+#include "LogfileTestNode.hpp"             // For all CPP Logfile API examples.
+
+using namespace std;
+
+using namespace polysync::datamodel;
+
+
+void LogFileTestNode::prepareLogfileToIterate()
+{
+    sleepMicro ( 1000000 ); // Sleep before mode off so queue doesn't flush.
+
+    _logFile->setModeOff(); // Automatically disables state.
+
+    _logFile->setFilePath( "/tmp/polysync_logfile.plog"  );
+
+    cout << "Logfile Iterator started." << endl;
+
+    _logFileWasIterated = true;
```

Comment:
Artifacts remaining from previous Logfile Example work setup. Fixed / removed this logic altogether in all Logfile Examples. https://github.com/PolySync/PolySync-CPP-Examples/pull/13/commits/4459cbb9d677792ef6e926bdda92d006acffea2b